### PR TITLE
changed: replace C99-style VLAs with std::vector

### PR DESCRIPTION
**Pull request by akva2**
thanks to @bska for pointing it out.

### CODE REVIEW

File: examples/upscale_cond.cpp [hunk 1]
```diff
@@ -963,8 +963,8 @@ try
        for (int idx=0; idx < points; ++idx) {
            if (node_vs_pressurepoint[idx] != 0) {
                // Receive data
-               double recvbuffer[2+tensorElementCount];
-               MPI_Recv(recvbuffer, 2+tensorElementCount, MPI_DOUBLE, 
+               std::vector<double> recvbuffer(2+tensorElementCount);
+               MPI_Recv(&recvbuffer[0], 2+tensorElementCount, MPI_DOUBLE,
```
Comment (atgeirr) [inline examples/upscale_cond.cpp]: Just a suggestion: we no longer need to support (very) ancient versions of gcc, so vector::data() is available.

**Comment by bska** [comment]
Good.

I **suppose** you could do
```C++
MPI_*(&v[0], v.size(), ...)
```
instead of the explicit expression for the `.size()`, but I'm not going to insist.

**Comment by akva2** [comment]
both suggestions incorporated.

**Comment by bska** [comment]
> both suggestions incorporated.

Great.  I'll merge into master before we drift too far away from the original context.

**Comment by atgeirr** [comment]
The process is quite quick from time to time...

**Comment by blattms** [comment]
On Tue, Jun 16, 2015 at 05:46:07AM -0700, Bård Skaflestad wrote:
> I **suppose** you could do
> ```C++
> MPI_*(&v[0], v.size(), ...)
> ```
> instead of the explicit expression for the `.size()`, but I'm not going to insist.

As long as the vector is never empty. If it is and one compile gcc's
stdlib debugging turned on, this will be a problem as you are indexing
an empty vector. (Bin there...)

**Comment by bska** [comment]
> As long as the vector is never empty. If it is and one compile gcc's stdlib debugging turned on, this will be a problem as you are indexing an empty vector. (Bin there...)

Good point.  In these cases we did ensure that the vectors were non-empty so that the expression `&v[0]` was meaningful.  We also switched to using `v.data()` which is always defined (`nullptr` if `v.empty()`) so the resulting call was valid regardless.  That said, the "indexing into empty on `&v[0]`" problem has caught me off guards on multiple occasions, so it is definitely something to be alert to.

**Pull request merged**